### ISSUE TITLE
FEATURE -  Support for custom transports

### ISSUE THREAD

**Issue by BearCooder** [mqttjs/MQTT.js#1616]
I am creating this feature request so it doesnt get lost from the conversation between @robertsLando & @vishnureddy17 in #1615 

**This library doesn't have support for custom transports.** 

The proprietary stuff has seeped into lots of areas in the code, it's not a standard transport protocol and there are specific to a proprietary product (e.g. WeChat specific stuff ). It is not sure if that belongs in an open source project. so the solution is either to tolerate this or add support for custom transport.

Because without a custom transport support all users using that protocol would have no alternatives.

Anyone that would like to pick this up for a future release? I can contribute and help probably in August.

**Comment by robertsLando** [comment]
This is already available now, the `MqttClient` instance is exported and can be used by providing a `streamBuilder` function like we do with transports now.

**Comment by BearCooder** [comment]
Awesome, so now in theory a cleanup of the proprietary stuff in the code could be done in combination with adding a note that this is removed and everyone can use via custom transports if needed?

**Comment by robertsLando** [comment]
> so now in theory a cleanup of the proprietary stuff in the code could be done in combination with adding a note that this is removed and everyone can use via custom transports if needed?

Yes, but based on a recent discussion about this we think removing some protocols support like ali and wx would not be a good idea, it would be better to place them somewhere so at least users that need them can find a way to use them, maybe a separete repository where we can put all propetary protocols `streamBuilder`
